type4py/vectorize.py

```diff
@@ -224,8 +231,27 @@ def process_datapoints(df, output_path, embedding_type, type, trans_func, cached
     if not os.path.exists(os.path.join(output_path, embedding_type + type + '_datapoints_x.npy')) or not cached_file:
         datapoints = df.apply(trans_func, axis=1)
 
-        datapoints_X = np.stack(datapoints.progress_apply(lambda x: x.generate_datapoint()),
-                                axis=0)
+        # optimize np.stack for datapoints in batches when handling large datasets
+        # emd_shape is based on the "identifiers" and "tokens"
+
+        # define batch sizes and rows for datapoints_x
+        batch_size = 1000
+        num_rows = datapoints.shape[0]
+
+        if embedding_type == "identifiers_":
+            emd_shape = 31
```

Comment:
Create a global variable in `__init__.py` for identifier length. E.g. `IDENTIFIER_SEQ_LEN`